executor/aggregate_xapi.go

```diff
@@ -77,6 +81,31 @@ func (n *finalAggregater) updateFirst(val types.Datum) error {
 	return nil
 }
 
+func (n *finalAggregater) updateMaxMin(val types.Datum, max bool) error {
+	ctx := n.getContext()
+	if val.IsNull() {
+		return nil
+	}
+	v := val.GetValue()
+	if !ctx.Evaluated {
```

Comment:
I think so. I will refactor aggregate related code in the next PR. It will be removed in the next PR.